src/main/java/de/tum/cit/aet/artemis/quiz/service/QuizExerciseService.java

```diff
@@ -291,31 +293,41 @@ public void handleDndQuizFileUpdates(QuizExercise updatedExercise, QuizExercise
             }
         }
 
-        fileService.deleteFiles(filesToRemove.stream().map(Path::of).toList());
+        var allFilesToRemoveMerged = filesToRemove.entrySet().stream()
+                .flatMap(entry -> entry.getValue().stream().map(p -> FilePathService.fileSystemPathForExternalUri(URI.create(p), entry.getKey()))).filter(Objects::nonNull)
```

Comment:
Even in streams I do not think that chars are a good practice as variable names
```suggestion
                .flatMap(entry -> entry.getValue().stream().map(path -> FilePathService.fileSystemPathForExternalUri(URI.create(path), entry.getKey()))).filter(Objects::nonNull)
```